src/main/java/edu/kit/compiler/codegen/pattern/UnknownPattern.java

```diff
@@ -0,0 +1,60 @@
+package edu.kit.compiler.codegen.pattern;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import edu.kit.compiler.codegen.MatcherState;
+import edu.kit.compiler.codegen.Operand;
+import edu.kit.compiler.codegen.Util;
+import edu.kit.compiler.intermediate_lang.Instruction;
+import firm.bindings.binding_irnode.ir_opcode;
+import firm.nodes.Node;
+import lombok.RequiredArgsConstructor;
+
+public final class UnknownPattern implements Pattern<InstructionMatch> {
+
+    @Override
+    public InstructionMatch match(Node node, MatcherState matcher) {
+        if (node.getOpCode() == ir_opcode.iro_Unknown) {
+            var size = Util.getSize(node.getMode());
```

Comment:
With the changes in #123, constant optimization will replace arithmetic / boolean operations with at least one unknown operand with a new `Unknown` node. In the following example, this leads to an `Unknown` node of mode `B` as the predecessor of the `Cond` node.

The problem: `Util.getSize` will fail for mode `B`.

Example:
```
class Test {
    public static void main(String[] args) {
        Test x;
        if (x == null) {}
    }
}
```
(similar to `CompareRefWithNull.valid.mj`)

Reply:
Hm, that is a problem,  I can't (and it wouldn't make sense to) allocate a register for a Firm boolean. We could replace `Cond(Unknown)` with an unconditional jump to either of the control flow successors.

Reply:
That would be a valid solution since `Cond(Unknown)` produces undefined behavior. Where would we implement this though?

Reply:
It needs to run for `-O0` as well, so currently either during constant propagation or arithmetic identities.

Reply:
My first thought is that it's neither a constant propagation nor an arithmetic identity, it's just resolving undefined behavior. But if it needs to be in one of the two, I guess it would fit better in constant propagation.

Reply:
We could obviously introduce an entirely new optimization, but that's another graph pass each iteration. Also I don't think it's entire misplaced in constant propagation. You're already propagating `Unknown`s for other kinds of nodes, so why not do it for `Cond` as well.

Reply:
Alright, I will include it in #123.